addons/mrp/models/stock_move.py

```diff
@@ -125,6 +125,16 @@ def _compute_is_done(self):
         for move in self:
             move.is_done = (move.state in ('done', 'cancel'))
 
+    @api.depends('raw_material_production_id.name')
+    def _compute_reference(self):
+        prod_move = self.filtered(lambda m: m.raw_material_production_id)
+        for move in prod_move:
+            move.write({
+                'name': move.raw_material_production_id.name,
+                'reference': move.raw_material_production_id.name,
+            })
+        super(StockMove, self - prod_move)._compute_reference()
```

Comment:
```suggestion
        not_prod_move = set()
        for move in self:
            if not move.raw_material_production_id:
                not_prod_move.add(move.id)
                continue
            move.write({
                'name': move.raw_material_production_id.name,
                'reference': move.raw_material_production_id.name,
            })
        super(StockMove, self.browse(not_prod_move))._compute_reference()
```
Always good to avoid an extra loop

Reply:
Done, thanks! But I think it's better to use an empty recordset to avoid doing a browse when calling the super, what do you think ?